app/src/main/java/com/alexbar/layoutbasic/screen/confirm_payment/screens/ConfirmPaymentScreen.kt

```diff
@@ -0,0 +1,124 @@
+
+    LaunchedEffect(Unit) {
+        scope.launch {
+            snackBarHostState
+                .showSnackbar(welcome_message)
+        }
+    }
+
+    ConstraintLayout(modifier = Modifier
+        .fillMaxSize()
+        .background(Color.White)
+    ) {
+        val (header, paymentMethodSegmentedControl, payButton, snackBar,
+            userBalance, emptyCreditCards) = createRefs()
+
+        HeaderConfirmPayment(
+            userBalance = total_amount,
+            modifier = Modifier.constrainAs(header){}
+        )
+
+        SegmentedControlConfirmPayment(
+            title = payment_method_title,
+            options = paymentMethods,
+            selectedIndex = selectedIndex,
+            onOptionSelected = { index -> selectedIndex = index },
+            modifier = Modifier.constrainAs(paymentMethodSegmentedControl) {
+                top.linkTo(header.bottom)
+            }
+        )
+
+        if (selectedIndex == 0) {
+            UserBalance(
+                currentBalance = currentUserBalance,
+                modifier = Modifier.constrainAs(userBalance) {
+                    top.linkTo(paymentMethodSegmentedControl.bottom)
+                }
+            )
+        } else {
+            EmptyCreditCards(
+                modifier = Modifier.constrainAs(emptyCreditCards) {
+                    top.linkTo(paymentMethodSegmentedControl.bottom)
+                },
+                onClickAddNewCard = { showNewCreditCardDialog = true }
+            )
+        }
+
+        if (showNewCreditCardDialog)
+            AddCreditCardDialog { showNewCreditCardDialog = false }
+
+        Button(
+            onClick = {
+                if (selectedIndex == 1)
+                    scope.launch {
+                        val result = snackBarHostState
+                            .showSnackbar(
+                                message = no_credit_cards_message,
+                                actionLabel = add_credit_cards_button,
+                                duration = SnackbarDuration.Short
+                            )
+                        when (result) {
+                            SnackbarResult.ActionPerformed -> {
+                                showNewCreditCardDialog = true
+                            }
+                            SnackbarResult.Dismissed -> {}
+                        }
+                    }
+            },
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(dimen_16_dp)
+                .constrainAs(payButton) {
+                    bottom.linkTo(parent.bottom)
+                    start.linkTo(parent.start)
+                    end.linkTo(parent.end)
+            }
+        ) {
+            Text(text = checkout_button)
+        }
+
+        SnackbarHost(
+            hostState = snackBarHostState,
+            modifier = Modifier.constrainAs(snackBar) {
+                bottom.linkTo(parent.bottom)
+            }
+        )
```

Comment:
@ezatpanah Is it appropriate to use a single SnackbarHost to show several SnackBars? I'm using it for both the welcome message and to inform the user that they don't have any credit cards

Reply:
Yes , we can use this approach to show several SnackBars, the example :
`
val snackbarHostState = remember { SnackbarHostState() }

SnackbarHost(
    hostState = snackbarHostState,
    modifier = Modifier.constrainAs(snackBar) {
        bottom.linkTo(parent.bottom)
    }
)

snackbarHostState.showSnackbar("your msg!")

snackbarHostState.showSnackbar("You don't have any credit cards.")`